### ISSUE TITLE
Highlighting of uploaded file doesn't work (anymore)

### ISSUE THREAD

**Issue by marcelrummens** [ckfinder/ckfinder#343]
In CKFinder 2.x the uploaded file was automatically highlighted. This feature seems to be missing in CKFinder 3.4.2. Or am I missing a configuration option?

If not, could this feature be re-implemented?

**Comment by zaak** [comment]
Hi @marcelrummens. I have just checked our [demo](https://ckeditor.com/ckeditor-4/ckfinder/) and uploaded file seems to be highlighted correctly:

![screenshot_2018-01-24_10-33-19](https://user-images.githubusercontent.com/803299/35324868-5cc9507a-00f2-11e8-91ca-27969063a56b.png)

Could you please check if  [demo](https://ckeditor.com/ckeditor-4/ckfinder/) works for you? If there's still any issue please provide more details about it (please have look at [issue tracker instructions](https://docs.ckeditor.com/ckfinder/ckfinder3/#!/guide/dev_feedback)).

**Comment by marcelrummens** [comment]
Hello @zaak 
After successfully testing it on the demo, I realised we have a custom plugin for renaming files to a custom format after uploading. This seems to be the culprit. This plugin renames the file after it has been uploaded (using the AFTER_COMMAND_FILE_UPLOAD event). I suppose CKFinder can't find the file under the original name when reloading the directory because it had been renamed.

Is it possible to change the filename before/during upload? If so, could you point me in the right direction? We're using the PHP connector.

**Comment by zaak** [comment]
Hey @marcelrummens. You are correct - if you rename the file in CKFinder connector plugin you have to inform the frontend about your change. If you have a plugin with `AFTER_COMMAND_FILE_UPLOAD` listener then it should be a matter of updating the `fileName` field in the JSON response. Please have a look at following code snippet:

```php
    public function onAfterFileUpload(AfterCommandEvent $event)
    {
       // Your custom file renaming logic
       // ...

        /** @var \CKSource\CKFinder\Response\JsonResponse $response */
        $response = $event->getResponse();

        $responseData = $response->getData();
        $responseData['fileName'] = $yourChangedFilenameHere; // Here pass the new file name
        $response->setData($responseData);
    }

    public static function getSubscribedEvents()
    {
        return [
            CKFinderEvent::AFTER_COMMAND_FILE_UPLOAD => 'onAfterFileUpload'
        ];
    }
```

**Comment by marcelrummens** [comment]
Works like a charm, thank you very much!

**Comment by zaak** [comment]
That's great! If you don't mind, I'll close this issue.

**Comment by marcelrummens** [comment]
Hi, I'm still having some problems with the highlighting, which are reproducible on the demo page.
When the list of files is long, the file is highlighted, but the window doesn't jump to it. I uploaded a file starting with z to test this.

Tested on Mac OS 10.13.2

Safari 11.0.2 (13604.4.7.1.6): Doesn't highlight file at all
Chrome 64.0.3282.119: Highlights file, but doesn't jump to it with a lot of files in the directory
Firefox 58.0.1: Highlights file, but doesn't jump to it with a lot of files in the directory
